src/helpers/approvals-satisfied.ts

```diff
@@ -80,15 +80,32 @@ export const approvalsSatisfied = async ({
 
     const numberOfRequiredReviews =
       teamOverrides?.find(({ team }) => team && entry.owners.includes(team))?.numberOfRequiredReviews ?? number_of_reviewers;
-    core.info(`Current number of approvals satisfied for ${entry.owners}: ${numberOfApprovals}`);
-    core.info(`Number of required reviews: ${numberOfRequiredReviews}`);
+    logs.push(`Current number of approvals satisfied for ${entry.owners}: ${numberOfApprovals}`);
+    logs.push(`Number of required reviews: ${numberOfRequiredReviews}`);
 
     return numberOfApprovals >= Number(numberOfRequiredReviews);
   };
 
-  core.info(`Required code owners: ${requiredCodeOwnersEntriesWithOwners.map(({ owners }) => owners).toString()}`);
+  logs.push(`Required code owners: ${requiredCodeOwnersEntriesWithOwners.map(({ owners }) => owners).toString()}`);
+
   const booleans = await Promise.all(requiredCodeOwnersEntriesWithOwners.map(codeOwnersEntrySatisfiesApprovals));
-  return booleans.every(Boolean);
+  const approvalsSatisfied = booleans.every(Boolean);
+
+  core.info(logs.join('\n'));
+
+  if (!approvalsSatisfied) {
+    logs.unshift('Required approvals not satisfied:\n');
```

Comment:
never heard of unshift until now, but it is equal to `['Required approvals not satisifed', ...logs]`